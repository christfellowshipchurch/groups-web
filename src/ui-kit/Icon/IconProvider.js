import React from 'react';
import PropTypes from 'prop-types';

// Create our IconContext for icon overrides
const IconContext = React.createContext({});

// Accept iconInput as a prop and pass it to the context provider
const IconProvider = ({ children, iconInput, ...props }) => (
  <IconContext.Provider value={iconInput} {...props}>
    {children}
  </IconContext.Provider>
);

IconProvider.propTypes = {
  children: PropTypes.node,
  iconInput: PropTypes.objectOf(PropTypes.func),
};

IconProvider.defaultProps = {
  iconInput: {},
};

export { IconContext, IconProvider as default };
