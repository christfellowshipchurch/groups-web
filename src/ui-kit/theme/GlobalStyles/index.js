import { createGlobalStyle } from 'styled-components';
import { normalize } from 'styled-normalize';

import { h2Style } from '../../typography/H2';
import { h3Style } from '../../typography/H3';
import { h4Style } from '../../typography/H4';
import { h5Style } from '../../typography/H5';
import { h6Style } from '../../typography/H6';

const GlobalStyles = createGlobalStyle`
  ${normalize}

  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }

  html {
    color: ${({ theme }) => theme.colors.text.primary};
    font-size: 100%;
    font-family: ${({ theme }) => theme.typography.fontFamily};
  }

  ::selection {
    background: ${({ theme }) => theme.colors.primary}
    text-shadow: none;
  }

  /* Individal typography styles are imported to and referenced from the global stylesheet. See
     typogrpahy components (/ui-kit/typography) for style definitions */
  ${h2Style}
  ${h3Style}
  ${h4Style}
  ${h5Style}
  ${h6Style}

  /* Copied from https://github.com/buildo/react-placeholder/blob/master/src/reactPlaceholder.scss
  to reduce technical complexity with importing css */
  .show-loading-animation {
    &.rect-shape,
    &.round-shape,
    &.text-row,
    &.custom-placeholder, /* We add a custom class for use in our placeholder components */
    .rect-shape,
    .round-shape,
    .text-row,
    .custom-placeholder { /* We add a custom class for use in our placeholder components */
      animation: react-placeholder-pulse 1.5s infinite;
    }
  }

  @keyframes react-placeholder-pulse {
    0% {
      opacity: .6;
    }
    50% {
      opacity: 1;
    }
    100% {
      opacity: .6;
    }
  }
`;

export default GlobalStyles;
