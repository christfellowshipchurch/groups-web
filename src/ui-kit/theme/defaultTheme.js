import { css } from 'styled-components';

const theme = {
  colors: {
    primary: 'salmon',
    text: {
      primary: '#303030',
      secondary: '#505050',
    },
    background: {
      paper: '#FFFFFF',
      inactive: '#A5A5A5',
    },
  },
  typography: {
    baseFontSize: 1,
    baseLineHeight: 1.5,
    fontFamily:
      'Inter, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", "Roboto", "Helvetica Neue", Arial, sans-serif',
  },
  sizing: {
    baseUnit: (unit = 1) => `${unit}rem`,
    baseBorderRadius: '16px',
    avatar: {
      small: '40',
      medium: '80',
      large: '120',
    },
  },
  shadows: {
    default: '0px 5px 18px rgba(0, 0, 0, 0.1)',
  },
  helpers: {
    /*
     * Vertical rhythm method of spacing objects (typically typographic elements) based on a predefined
     * ratio. A popular ratio that is often used would be something like the Golden Ratio. With the
     * Golden Ratio as an example, all elment spacing on the page is a multiplied derivitive of that
     * ratio resulting in a rhthmic spacing. This is often used in typography to set line heights and
     * vertical paddings. The helper function below takes a `rem` unit (often a font size but not always)
     * and multiplies it by either a custom ratio or a derrived ratio from the themes `baseLineHeight`
     * devided by the `baseFontSize`.
     */
    verticalRhythm: (units, customRatio) => {
      const verticalRatio =
        customRatio ||
        theme.typography.baseLineHeight / theme.typography.baseFontSize;
      return `${verticalRatio * units}rem`;
    },
  },
  overrides: {
    'uiKit.typography.H1.StyledH1': {
      $style: {
        background: 'salmon',
      },
      love: 'Jesus',
    },
    'uiKit.typography.H2.StyledH2': {
      $style: css`
        background: goldenrod;
      `,
      love: 'God',
    },
  },
};

export default theme;
