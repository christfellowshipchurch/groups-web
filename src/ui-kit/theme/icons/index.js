export ArrowBack from './ArrowBack.svg';
export ArrowDown from './ArrowDown.svg';
export ArrowNext from './ArrowNext.svg';
export ArrowUp from './ArrowUp.svg';
export Audio from './Audio.svg';
export Avatar from './Avatar.svg';
export Bank from './Bank.svg';
export BrandIcon from './BrandIcon.svg';
export Building from './Building.svg';
export Calendar from './Calendar.svg';
export Camera from './Camera.svg';
export Check from './Check.svg';
export CheckboxChecked from './CheckboxChecked.svg';
export CheckboxUnchecked from './CheckboxUnchecked.svg';
export CircleOutlineCheckMark from './CircleOutlineCheckMark.svg';
export CircleOutlineXMark from './CircleOutlineXMark.svg';
export CircleOutlinePlus from './CircleOutlinePlus.svg';
export Close from './Close.svg';
export Credit from './Credit.svg';
export Chromecast from './Chromecast.svg';
export Download from './Download.svg';
export Dropdown from './Dropdown.svg';
export Empty from './Empty.svg';
export Failed from './Failed.svg';
export Filter from './Filter.svg';
export Fullscreen from './Fullscreen.svg';
export Groups from './Groups.svg';
export Home from './Home.svg';
export Information from './Information.svg';
export Like from './Like.svg';
export LikeSolid from './LikeSolid.svg';
export LiveDot from './LiveDot.svg';
export Locate from './Locate.svg';
export Lock from './Lock.svg';
export Mute from './Mute.svg';
export Pause from './Pause.svg';
export Pin from './Pin.svg';
export Play from './Play.svg';
export PlayOpaque from './PlayOpaque.svg';
export PlaySolid from './PlaySolid.svg';
export Plus from './Plus.svg';
export Pray from './Pray.svg';
export Print from './Print.svg';
export Profile from './Profile.svg';
export RadioChecked from './RadioChecked.svg';
export RadioUnchecked from './RadioUnchecked.svg';
export Repeat from './Repeat.svg';
export RepeatAll from './RepeatAll.svg';
export RepeatOne from './RepeatOne.svg';
export Search from './Search.svg';
export Sections from './Sections.svg';
export Settings from './Settings.svg';
export Share from './Share.svg';
export Shuffle from './Shuffle.svg';
export SkipForwardThirty from './SkipForwardThirty.svg';
export SkipBackThirty from './SkipBackThirty.svg';
export SkipNext from './SkipNext.svg';
export SkipPrevious from './SkipPrevious.svg';
export Text from './Text.svg';
export Time from './Time.svg';
export Umbrella from './Umbrella.svg';
export Video from './Video.svg';
export VideoOff from './VideoOff.svg';
export Volume from './Volume.svg';
export Warning from './Warning.svg';
