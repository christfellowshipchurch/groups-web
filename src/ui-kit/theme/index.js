export defaultTheme from './defaultTheme';
export GlobalStyles from './GlobalStyles';
