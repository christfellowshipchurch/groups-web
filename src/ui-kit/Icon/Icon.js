import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
// importing from lodash this way allows for tree shaking!
import camelCase from 'lodash/camelCase';
import flow from 'lodash/flow';
import kebabCase from 'lodash/kebabCase';
import upperFirst from 'lodash/upperFirst';

import { getIsLoading } from '../isLoading';
import { Media } from '../Placeholder';
import * as uikitIcons from '../theme/icons';

import { IconContext } from './IconProvider';

/* Convenience component to render icons based on the icon's string name, like:
 *
 * <Icon name="skip-next" {...allOtherPropsPassedToComponent} />
 *
 * Can also import the icon directly:
 *
 * import { SkipNext } from '../theme/icons';
 * <SkipNext /> */

const IconComponent = styled(uikitIcons.Umbrella)`
  fill: ${({ theme }) => theme.colors.text.primary};
  height: auto;
  width: ${({ $size, theme }) =>
    typeof $size === 'string' // this allows size to be a percent if necessary
      ? $size
      : theme.sizing.baseUnit($size)};
`;

const pascalCase = (string) => flow(camelCase, upperFirst)(string);

const Icon = getIsLoading(({ name, isLoading, size, ...otherProps }) => {
  const iconInput = useContext(IconContext);

  const Icons = { ...uikitIcons, ...iconInput };

  return (
    <Media size={size} isLoading={isLoading}>
      <IconComponent
        as={Icons[pascalCase(name)]}
        $size={size}
        {...otherProps}
      />
    </Media>
  );
});

// eslint-disable-next-line consistent-return
const namePropValidator = (props, propName, componentName) => {
  const icons = Object.keys({ ...uikitIcons, ...props.iconInput }).map(
    kebabCase
  );

  if (!icons.includes(props.name)) {
    // eslint-disable-next-line no-console
    return new Error(
      `Invalid prop \`${propName}\` of value \`${
        props.name
      }\` supplied to \`${componentName}\` expected one of ${JSON.stringify(
        icons
      )}`
    );
  }
};

Icon.propTypes = {
  name: namePropValidator,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  fill: PropTypes.string,
  isLoading: PropTypes.bool,
};

Icon.defaultProps = {
  size: 2, // 2 === 2 baseUnit === 2rem === 32 is the default size used within the svg component.
};

Icon.displayName = 'Icon';

export default Icon;
