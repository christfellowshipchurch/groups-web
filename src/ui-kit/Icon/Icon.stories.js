import React from 'react';
import { storiesOf } from '@storybook/react';

import { kebabCase } from 'lodash';

import * as icons from '../theme/icons';

import Icon from '.';

const stories = storiesOf('ui-kit/Icon', module);

Object.keys(icons).forEach((iconName) => {
  stories.add(iconName, () => <Icon name={kebabCase(iconName)} />);
});

stories.add('isLoading', () => <Icon name={'umbrella'} isLoading />);
