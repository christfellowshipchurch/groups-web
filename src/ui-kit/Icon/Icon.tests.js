import React from 'react';
import renderer from 'react-test-renderer';
import { kebabCase } from 'lodash';
import 'jest-styled-components';

import Providers from '../../Providers';

import * as icons from '../theme/icons';

import Icon from '.';

Object.keys(icons).forEach((iconName) => {
  describe(`The ${iconName} icon`, () => {
    it('should render', () => {
      const tree = renderer.create(
        <Providers>
          <Icon name={kebabCase(iconName)} />
        </Providers>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
