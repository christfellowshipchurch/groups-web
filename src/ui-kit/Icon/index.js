import Icon from './Icon';

export IconProvider, { IconContext } from './IconProvider';

export default Icon;
