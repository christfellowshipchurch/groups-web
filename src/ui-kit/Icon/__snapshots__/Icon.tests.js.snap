// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ArrowBack icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The ArrowDown icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The ArrowNext icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The ArrowUp icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Audio icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Avatar icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Bank icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The BrandIcon icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Building icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Calendar icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Camera icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Check icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The CheckboxChecked icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The CheckboxUnchecked icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Chromecast icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The CircleOutlineCheckMark icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The CircleOutlinePlus icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The CircleOutlineXMark icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Close icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Credit icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Download icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Dropdown icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Empty icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Failed icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Filter icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Fullscreen icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Groups icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Home icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Information icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Like icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The LikeSolid icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The LiveDot icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Locate icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Lock icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Mute icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Pause icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Pin icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Play icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The PlayOpaque icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The PlaySolid icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Plus icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Pray icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Print icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Profile icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The RadioChecked icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The RadioUnchecked icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Repeat icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The RepeatAll icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The RepeatOne icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Search icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Sections icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Settings icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Share icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Shuffle icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The SkipBackThirty icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The SkipForwardThirty icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The SkipNext icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The SkipPrevious icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Text icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Time icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Umbrella icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Video icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The VideoOff icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Volume icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;

exports[`The Warning icon should render 1`] = `
.c0 {
  fill: #303030;
  height: auto;
  width: 2rem;
}

<Svgr
  className="c0"
/>
`;
